sdk/core/core-rest-pipeline/src/xhrHttpClient.ts

```diff
@@ -67,7 +67,13 @@ class XhrHttpClient implements HttpClient {
     for (const [name, value] of request.headers) {
       xhr.setRequestHeader(name, value);
     }
-    xhr.responseType = request.streamResponseStatusCodes?.size ? "blob" : "text";
+
+    xhr.responseType =
+      // Value of POSITIVE_INFINITY in streamResponseStatusCodes is considered as any status code
```

Comment:
nit: here it should be covered by `?.size` already.  Before sending request via xhr we have to specify response type. Here's just a heuristic for the type.